libnmstate/netapplier.py

```diff
@@ -201,6 +234,47 @@ def _verify_change(desired_state):
     desired_state.verify_dns(current_state)
 
 
+def _create_vlan_desired_state_from_vlan_current_state(
+    vlan_dstate, vlan_cstate, ifname
+):
+    return {
+        schema.Interface.NAME: ifname,
+        schema.Interface.MTU: vlan_dstate.get(schema.Interface.MTU)
+        or vlan_cstate[schema.Interface.MTU],
+        schema.Interface.TYPE: vlan_dstate.get(schema.Interface.TYPE)
+        or vlan_cstate[schema.Interface.TYPE],
+        schema.Interface.STATE: vlan_dstate.get(schema.Interface.STATE)
+        or vlan_cstate[schema.Interface.STATE],
+        schema.InterfaceType.VLAN: vlan_dstate.get(schema.InterfaceType.VLAN)
+        or vlan_cstate[schema.InterfaceType.VLAN],
+    }
+
+
+def _prepare_vlan_ifaces_for_mtu_change(desired_state):
+    current_state = state.State(netinfo.show())
```

Comment:
AFAICS you could pass the current_state from the caller function to ensure that all functions operate on the same current state.

Reply:
Sure, done! Thanks!

Reply:
Done! It have been fixed. Thanks!